cms/static/js/spec/video/transcripts/videolist_spec.js

```diff
@@ -215,41 +233,35 @@ function ($, _, Utils, VideoList, MetadataModel, AbstractEditor, AjaxHelpers) {
                         }
                     ];
 
-                spyOn(view, 'getVideoObjectsList').andReturn(videoListLength);
+                spyOn(view, 'getVideoObjectsList').and.returnValue(videoListLength);
                 spyOn(view, 'openExtraVideosBar');
 
-                waitsForResponse(
-                    mockServer,
-                    function () {
-                        assertToHaveBeenRendered(videoListLength);
-                        view.getVideoObjectsList.andReturn(videoListLength);
-                        expect(view.openExtraVideosBar).toHaveBeenCalled();
-                    },
-                    function () {
-                        resetSpies(mockServer);
-                        view.render();
-                    }
-                );
+                resetSpies(this.mockServer);
+                view.render();
 
-                waitsForResponse(
-                    mockServer,
-                    function () {
-                        assertToHaveBeenRendered(videoListHtml5mode);
+                waitsForResponse(this.mockServer)
+                    .then(function () {
+                        assertToHaveBeenRendered(videoListLength);
+                        view.getVideoObjectsList.and.returnValue(videoListLength);
                         expect(view.openExtraVideosBar).toHaveBeenCalled();
-                    },
-                    function () {
-                        resetSpies(mockServer);
-                        view.openExtraVideosBar.reset();
-                        view.getVideoObjectsList.andReturn(videoListHtml5mode);
+                    })
+                    .then(_.bind(function () {
+                        resetSpies(this.mockServer);
+                        view.openExtraVideosBar.calls.reset();
+                        view.getVideoObjectsList.and.returnValue(videoListHtml5mode);
                         view.render();
-                    }
-                );
 
+                        return waitsForResponse(this.mockServer)
+                            .then(function () {
+                                assertToHaveBeenRendered(videoListHtml5mode);
+                                expect(view.openExtraVideosBar).toHaveBeenCalled();
+                            });
+                    }, this))
+                    .then(done);
```

Comment:
`done()` needs to be called at line 258. Otherwise the spec will be marked as complete without waiting for the `waitForResponse()` on line 254.